### PR TITLE
Add getClusters feature

### PR DESCRIPTION
**Pull request by cambelem**
### Description
This feature introduces a new command to retrieve a list of Cluster attributes based on `organization_id`. 

### Key Features
- Defaults Output: JSON format.
- Added `--first` flag for the first cluster in the list
- Added searchStr for getClusters

### Usage Examples:

Input:
`ocm-support get clusters [org_id]`

Result:
```
[
  {
    "id": "1 ... a",
    "href": "/api/clusters_mgmt/v1/clusters/[cluster_id]",
    "name": "cluster-name1",
    "external_id": "1 ... c"
    "display_name": "cluster-name1",
    "subscription_id": "1 ... b",
    "state": "ready",
    "cloud_provider": "aws",
    "version": "4.16.0",
    "region_id": "us-east-1",
    "multi_az": true,
    "product_id": "rosa",
    "managed": true,
    "console_url": "https://console-openshift-console.apps.cluster-name1.com",
    "creation_timestamp": "2024-07-01T01:00:00.000000Z",
  },
...
]
```

Input:
`ocm-support get clusters [org_id] --first`

Result:
```
[
  {
    "id": "1 ... a",
    "href": "/api/clusters_mgmt/v1/clusters/[cluster_id]",
    "name": "cluster-name1",
    "external_id": "1 ... c"
    "display_name": "cluster-name1",
    "subscription_id": "1 ... b",
    "state": "ready",
    "cloud_provider": "aws",
    "version": "4.16.0",
    "region_id": "us-east-1",
    "multi_az": true,
    "product_id": "rosa",
    "managed": true,
    "console_url": "https://console-openshift-console.apps.cluster-name1.com",
    "creation_timestamp": "2024-07-01T01:00:00.000000Z",
  }
]
```

Input:
`ocm-support get clusters [org_id] "region.id='us-east-1'"`

Result:
```
[
  {
    "id": "1 ... a",
    "href": "/api/clusters_mgmt/v1/clusters/[cluster_id]",
    "name": "cluster-name1",
    "external_id": "1 ... c"
    "display_name": "cluster-name1",
    "subscription_id": "1 ... b",
    "state": "ready",
    "cloud_provider": "aws",
    "version": "4.16.0",
    "region_id": "us-east-1",
    "multi_az": true,
    "product_id": "rosa",
    "managed": true,
    "console_url": "https://console-openshift-console.apps.cluster-name1.com",
    "creation_timestamp": "2024-07-01T01:00:00.000000Z",
  },
...
]
```

### CODE REVIEW

File: cmd/ocm-support/get/clusters/cmd.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+)
+
+var args struct {
+	first bool
+}
+
+// Define the command structure
+var CmdGetClusters = &cobra.Command{
+	Use:     "clusters [org_id]",
+	Aliases: utils.Aliases["clusters"],
```
Comment (cristianoveiga) [inline cmd/ocm-support/get/clusters/cmd.go:19]: In addition to the org_id, I think we should support the same parameters as subscriptions:

 [id|external_id|organization_id] [optional_search_string]

**Comment by tirthct** [comment]
In addition to these file changes, we can also update the readme with a section about this command

File: cmd/ocm-support/get/clusters/cmd.go (hunk 1 above)
Comment (cambelem) [reply cmd/ocm-support/get/clusters/cmd.go:19]: Updated it to use:
`clusters [id|external_id|organization_id|subscription_id] [optional_search_string]`

- id = cluster_id
- external_id = cluster_external_id

File: pkg/cluster/cluster.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+	// Access the clusters API
+	collection := connection.ClustersMgmt().V1().Clusters()
+
+	// Fetch clusters
+	page := 1
+	pageSize := limit
+	if limit <= 0 || limit > 100 {
+		pageSize = 100
+	}
```
[the remark this answers is not in the document]
Comment (cambelem) [reply pkg/cluster/cluster.go:51]: Hmm, do we want to go away from pagination for this?

Subscripions does the following:
```
	if len(subscriptions) > utils.MaxRecords {
		return fmt.Errorf("too many (%d) subscriptions found. Consider changing your search criteria to something more specific by providing optional search filters as a second argument", len(subscriptions))
	}
```

Which will flat error with "too many". My way should allow the user to pull as many clusters as they want. Though that piece of code you highlighted has a problem if the user wants the first 150 (it'll always return 100).

**Comment by cristianoveiga** [comment]
LGTM

**Pull request merged**